### PR TITLE
Add --generate-mesh-id-experimental flag to server bootstrap

### PR DESCRIPTION
**Pull request by stanley-cheung**
This flag has been added to the client deployment but not the server deployment. Adding it to server now.

We need this fix in order for the CSM Observability Test to properly record the server side metrics.

### CODE REVIEW

File: tests/gamma/csm_observability_test.py [hunk 1]
```diff
@@ -50,7 +50,8 @@ def test_csm_observability(self):
         #   resource creation out of self.startTestServers()
         with self.subTest("1_run_test_server"):
             test_server: _XdsTestServer = self.startTestServers(
-                enable_csm_observability=True
+                enable_csm_observability=True,
```
Comment (gnossen) [inline tests/gamma/csm_observability_test.py:53]: Should we add this to an override of [`RegularXdsKubernetesTestCase.startTestServers`](https://github.com/grpc/psm-interop/blob/292cea23d6263862416008b0a76b4c0c16cbaf69/framework/xds_k8s_testcase.py#L834) within [`GammaXdsKubernetesTestCase`](https://github.com/grpc/psm-interop/blob/292cea23d6263862416008b0a76b4c0c16cbaf69/framework/xds_gamma_testcase.py#L35), similar to how `startTestClient` currently has a GAMMA-specific override. This way, all future CSM tests will get this and we won't have to go through this debugging process again in the future.

Comment (sergiitk) [reply tests/gamma/csm_observability_test.py:53]: Good point - but do we actually need to set `generate_mesh_id` in all gamma tests? If so, then we should make `GammaXdsKubernetesTestCase.startTestServers()` to set it automatically. If we don't (f.e. in affinity / baseline) tests, I think we should keep it as is.

Comment (gnossen) [reply tests/gamma/csm_observability_test.py:53]: Yes. We do need `generate_mesh_id` in all GAMMA tests.

Comment (stanley-cheung) [reply tests/gamma/csm_observability_test.py:53]: Ok, will try to refactor. Thanks for the confirmation.

**Comment by sergiitk** [comment]
- [ ] [grpc/core/master/linux/psm-csm](https://source.cloud.google.com/results/invocations/8665b61c-4627-41cf-b0bb-71fd08689694)

**Review by sergiitk — APPROVED** [review]
Refactoring done, all good after the test passes

**Pull request merged**